inject/maven-plugin/src/main/java/io/helidon/inject/maven/plugin/AbstractCreatorMojo.java

```diff
@@ -65,80 +43,34 @@ public abstract class AbstractCreatorMojo extends AbstractMojo {
-
-    /**
-     * The file name written to ./target/inject/ to track the last package name generated for this application.
-     * This application package name is what we fall back to for the application name and the module name if not otherwise
-     * specified directly.
-     */
-    protected static final String APPLICATION_PACKAGE_FILE_NAME = ModuleUtils.APPLICATION_PACKAGE_FILE_NAME;
-
     // ----------------------------------------------------------------------
     // Configurables
     // ----------------------------------------------------------------------
-
     /**
-     * The template name to use for codegen.
-     */
-    @Parameter(property = TemplateHelper.TAG_TEMPLATE_NAME, readonly = true, defaultValue = TemplateHelper.DEFAULT_TEMPLATE_NAME)
-    private String templateName;
-
-    /**
-     * The module name to apply. If not found the module name will be inferred.
+     * The module name to apply. If not found the module name will be inferred
+     * from {@code module-info.java} if present, or defined as {@code unnamed/package name}.
+     *
+     * @see io.helidon.codegen.CodegenOptions#TAG_CODEGEN_MODULE
      */
-    @Parameter(property = Options.TAG_MODULE_NAME, readonly = true)
+    @Parameter(property = CodegenOptions.TAG_CODEGEN_MODULE)
     private String moduleName;
-
-    // ----------------------------------------------------------------------
-    // Generic Configurables
-    // ----------------------------------------------------------------------
-
     /**
-     * The current project instance. This is used for propagating generated-sources paths as
-     * compile/testCompile source roots.
+     * The package name to apply. If not found the package name will be inferred.
+     *
+     * @see io.helidon.codegen.CodegenOptions#TAG_CODEGEN_PACKAGE
      */
-    @Parameter(defaultValue = "${project}", readonly = true, required = true)
-    private MavenProject project;
-
+    @Parameter(property = CodegenOptions.TAG_CODEGEN_PACKAGE)
+    private String packageName;
     /**
      * Indicates whether the build will continue even if there are compilation errors.
      */
     @Parameter(property = TAG_FAIL_ON_ERROR, defaultValue = "true")
-    private boolean failOnError = true;
-
+    private boolean failOnError;
     /**
      * Indicates whether the build will continue even if there are any warnings.
      */
     @Parameter(property = TAG_FAIL_ON_WARNING)
     private boolean failOnWarning;
-
-    /**
-     * The -source argument for the Java compiler.
-     * Note: using the same as maven-compiler for convenience and least astonishment.
-     */
-    @Parameter(property = "maven.compiler.source", defaultValue = DEFAULT_SOURCE)
-    private String source;
-
-    /**
-     * The -target argument for the Java compiler.
-     * Note: using the same as maven-compiler for convenience and least astonishment.
-     */
-    @Parameter(property = "maven.compiler.target", defaultValue = DEFAULT_TARGET)
```

Comment:
The place you were referring to "21" should be changed to something like this using `DEFAULT_TARGET`

Reply:
Why? There is not benefit in introducing a constant if it is used only once and never referenced. It is actually harder to read, as you need to go to the constant to find out the value.